packages/ui/src/components/sign-in/sign-in.tsx

```diff
@@ -204,7 +205,11 @@ export function SignInComponentLoaded() {
                       <Card.Description>{t('signIn.password.subtitle')}</Card.Description>
                       <Card.Description>
                         <span className='flex items-center justify-center gap-2'>
-                          <SignIn.SafeIdentifier />
+                          <SignIn.SafeIdentifier
+                            transform={val => {
+                              return formatSafeIdentifier(val) || val;
```

Comment:
If we add some overloads to the `formatSafeIdentifier` function to tell TypeScript that it will always return a string if provided a string, we can remove this logical OR which won't ever be invoked since `SafeIdentifier` falls back to an empty string in the worse case.

[TS Playground](https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABMOAnAtgQygZU8AUwEkATAsWYGA1ACgGcpUAuRMEAGw4EpX2uA3AChQkWAmRosufMTIUYVGgyatwZKmAIleidQU3bho6PCQoM2PIVLlK1OoxaInMMAHNdrj8fCmJFtLWcnaKDirO3u6IAD5snByxemAabtrciADeQoi5iIqItACETrFxTgB0bhAcIGT0tADkAAKNZS5MFdIQABa0APQA2pgAtABeAIIjAFoAuv3ci1k5eauoBFAgqEhOwqsAvkIrueub24iNjcKHQhAIjPn0EzhMbtEAvJKWMjby9sqXbjCO5gB4wegAOQSiE+gSsslsCiUdH4PGB9ygjwAqikDGkSDCvkEEX8wsp9IYdAIgA)